lib/ecto_boot_migration.ex

```diff
@@ -73,121 +72,152 @@ defmodule EctoBootMigration do
+    log("Loading application #{inspect(app)}...")
 
-      migrations = []
+    if loaded?(app) do
+      start_dependencies()
+      repos = Application.get_env(app, :ecto_repos, [])
+      repos_pids = start_repos(repos)
+      migrations = run_migrations(repos)
+      stop_repos(repos_pids)
 
+      log("Done")
       case migrations do
         [] ->
           {:ok, :noop}
 
         migrations ->
           {:ok, {:migrated, migrations}}
       end
-
-    else 
+    else
       {:error, :not_loaded}
     end
   end
 
+  def loaded?(app) do
+    case Application.load(app) do
+      :ok ->
+        log("Loaded application #{inspect(app)}")
+        true
+
+      {:error, {:already_loaded, ^app}} ->
+        log("Application #{inspect(app)} is already loaded")
+        true
+
+      {:error, reason} ->
+        log("Failed to start the application: reason = #{inspect(reason)}")
+        false
+    end
+  end
+
+  @doc """
+  Start the Repo(s) for app, returns pids
+  """
+  def start_repos(repos) do
+    log("Starting repos...")
+
+    repos_pids =
+      repos
+      |> Enum.reduce([], fn repo, acc ->
+        log("Starting repo: #{inspect(repo)}")
+
+        case repo.start_link(pool_size: 1) do
+          {:ok, pid} ->
+            log("Started repo: pid = #{inspect(pid)}")
+            [pid | acc]
+
+          {:error, {:already_started, pid}} ->
+            log("Repo was already started: pid = #{inspect(pid)}")
+            acc
+
+          {:error, reason} ->
+            log("Failed to start the repo: reason = #{inspect(reason)}")
+            acc
+        end
+      end)
+
+    log("Started repos, pids = #{inspect(repos_pids)}")
+    repos_pids
+  end
+
+  def run_migrations(repos) do
+    log("Running migrations")
+
+    migrations =
+      repos
+      |> Enum.reduce([], fn repo, acc ->
+        log("Running migration on repo #{inspect(repo)}")
+
+        result = Ecto.Migrator.run(repo, migrations_path(repo), :up, all: true)
+        log("Run migration on repo #{inspect(repo)}: result = #{inspect(result)}")
+        acc ++ result
+      end)
+
+    log("Run migrations: count = #{length(migrations)}")
+    migrations
+  end
+
+  @doc """
+  Start apps necessary for executing migrations
+  """
+  def start_dependencies do
+    log("Starting dependencies...")
+
+    @apps
+    |> Enum.each(fn app ->
+      log("Starting dependency: application #{inspect(app)}")
+      Application.ensure_all_started(app)
+    end)
+
+    log("Started dependencies")
+  end
+
+  def stop_repos(repos_pids) do
+    log("Cleaning up...")
+    log("Stopping repos...")
+
+    repos_pids
+    |> Enum.each(fn repo_pid ->
+      log("Stopping repo #{inspect(repo_pid)}...")
+      Process.exit(repo_pid, :normal)
+      log("Stopped repo #{inspect(repo_pid)}")
+    end)
+
+    log("Stopped repos")
+    log("Cleaned up")
+  end
+
+  def log(msg), do: log(msg, debug?())
+  def log(msg, true), do: IO.puts("[EctoBootMigration] #{msg}")
```

Comment:
Why not use `Logger.debug` instead of `IO.puts?`

Reply:
Hello @scarfacedeb, thank for the question! 

I don't like the super-verbose output any time I start my Phoenix application (even in dev mode). Same for running tests. It is only useful for me when troubleshooting `ecto_boot_migrations`. So normally I would like to silence the logging 100%. Logger module (https://hexdocs.pm/logger/Logger.html) provides much more flexibility and configuration parameters that not needed for this particular situation. 

So, that is why I opted in for a very barebones way to log debugging statements with a simple on/off config param.

Reply:
I think it's possible to keep your silence config and use `Logger.debug` at the same time.

I prefer to use Logger in such cases, because it's more predictable (most other packages use `Logger` to log messages), enables configuration, allows `capture_log` in tests.

e.g. you could configure `Logger` to use other logging backends and it would be weird if all of the logs will go there, except `ecto_boot_migrations` logs.

Reply:
In any case, I hope it'll be merged soon. :v: